currency/manager.go

```diff
@@ -458,6 +458,33 @@ func (p *PairsManager) getPairStoreRequiresLock(a asset.Item) (*PairStore, error
 	return pairStore, nil
 }
 
+// SetDelimitersFromConfig ensures that the pairs adhere to the configured delimiters
+// Pairs.Unmarshal doesn't know what the delimiter is, so uses the first punctuation rune
+func (p *PairsManager) SetDelimitersFromConfig() error {
+	for a, s := range p.Pairs {
+		cf := s.ConfigFormat
+		if cf == nil {
+			cf = p.ConfigFormat
+		}
+		if cf == nil {
+			continue
+		}
+		for i, p := range []Pairs{s.Enabled, s.Available} {
+			for j := range p {
+				if p[j].Delimiter == cf.Delimiter {
+					continue
+				}
+				nP, err := NewPairDelimiter(p[j].String(), cf.Delimiter)
+				if err != nil {
+					return fmt.Errorf("%s.%s.%s: %w", a, []string{"enabled", "available"}[i], p[j], err)
+				}
+				p[j] = nP
+			}
+		}
+	}
+	return nil
+}
```

Comment:
This is nice to highlight things, but its placement, where an error leads to GCT being unable to start could be an issue such as the errors raised from `NewPairDelimiter`.
For example, ITBIT has an incorrect config format set in `itbit_wrapper.go:64` with the inclusion of a `DashDelimiter`. However, no currencies in the config contain one, which causes `NewPairDelimiter` to fail, and GCT unable to start. Removing it from the config will only re-add it on next startup, reestablishing the issue. To reproduce this yourself, enable ITBIT and run GCT

For this PR to be merged, you will need to ensure that all set config formats work for all exchanges and their assets or consider calling `SetDelimitersFromConfig` from a place which doesn't impact the entire startup of the application. If a user has an encrypted config, they will have to start again by running this because they can't modify their config formats to fix the fatal error on startup.

Reply:
Understood.

I considered whether this should be a fatal startup issue or not, and kept coming back to "It should be" because otherwise we have a fundamental config issue and it's too easily ignored on startup.

Yes, It was my intention to fix any issues arising from this in config_example.json and testconfig.json.

Regarding ITBIT: I didn't/don't get an issue, because config_example.json has:
```
    "configFormat": {
     "uppercase": true,
     "delimiter": "-"
    },
```
So the SetDefault version doesn't get a look-in.
Best shot I get is when I turn it's `useGlobalFormat` to false, and then I get:
`2023/12/12 15:30:18 Unable to initialise bot engine. Error: failed to load config. Err: fatal error checking config values. Error: no exchanges enabled`

Could you share the config I can use to replicate what you see?

In the meantime, I can also just fix the itbit_wrapper, right?

Your last comment is more worrying, I guess. I'm not sure how they'd go about fixing any other issues with config either, not having experimented with it.

I'm happy to switch the fatal error to a warning, which would mitigate the issues, but leave us more vulnerable.
And/Or We could disable the exchange like we do in other situations.

Reply:
ITBit: I couldn't reproduce because this issue happens on the *second* run, after the config is updated.
I was running with -dryrun so never got the config update. 🤦 

Fixed: 45b9d362

Reply:
We've discussed this and we're agreed now that whilst erroring isn't great, it's the necessary evil here.
Resolving.

I'm going to ensure that Binance is okay with it's [special handling](https://github.com/thrasher-corp/gocryptotrader/blob/master/exchanges/binance/binance_wrapper.go#L313-L360)